embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/comms/delivery/v2/CacheService2.kt

```diff
@@ -0,0 +1,8 @@
+package io.embrace.android.embracesdk.internal.comms.delivery.v2
+
+import java.io.File
+
+interface CacheService2 {
+    fun deletePayload(file: File)
```

Comment:
Is it worth creating an abstraction over `File`, such as `Payload`, that contains the internal implementation that loads a file for sending?

One of the problems with our current approach is that it's fairly simple to just delete or transform a `File` once a reference is obtained & this breaks assumptions made by other components. I can't decide whether this is over-engineering or justified paranoia so would be interesting to hear your thoughts

Reply:
Oh yeah, we should do that. I just wanted something quick here but it should be a durable way to uniquely referencing a payload. Frankly, the file name is probably good enough if we expect the file name no never change, i.e. the timestamp is always when it was first written or something.